### PR TITLE
docs(dh): add docs for feature flags

### PR DESCRIPTION
**Pull request by frontend-specialisten**
## Description

Add docs for feature flags

## References

- #408

### CODE REVIEW

File: libs/dh/shared/feature-flags/README.md [hunk 1]
```diff
@@ -1 +1,60 @@
...
+```ts
+'example-feature-flag': {
+    created: '01-01-2022',
+    disabledEnvironments: [DhAppEnvironment.prod],
+  }
+```
+
+The `created` property is danish locale `dd-mm-yyyy`. Feature flags use this property internally to check for old feature flags.
+
+The `disabledEnvironments` property is an array of the environments you wan't to disable the feature, for ex., `prod.`
```
Comment (dzhavat) [inline libs/dh/shared/feature-flags/README.md:26]: suggestion:

```suggestion
The `disabledEnvironments` property is an array of the environments you want to disable a feature in, e.g. `prod`.
```

File: libs/dh/shared/feature-flags/README.md [hunk 2]
```diff
@@ -1 +1,60 @@
...
+The feature flags require two properties, `created` and `disabledEnvironments`.
+
+**Example:**
+
+```ts
+'example-feature-flag': {
+    created: '01-01-2022',
+    disabledEnvironments: [DhAppEnvironment.prod],
+  }
+```
```
Comment (dzhavat) [inline libs/dh/shared/feature-flags/README.md:22]: nitpick:suggestion: let's use the same feature flag name mentioned later in the doc.

```suggestion
```ts
'my-awesome-feature-flag': {
  created: '01-01-2022',
  disabledEnvironments: [DhAppEnvironment.prod],
}
```
```

**Review by ManBearTM — COMMENTED** [review]
Awesome! I just have one question 🙂

File: libs/dh/shared/feature-flags/README.md [hunk 3]
```diff
@@ -1 +1,59 @@
...
+
+The `created` property is danish locale `dd-mm-yyyy`. Feature flags use this property internally to check for old feature flags.
+
+The `disabledEnvironments` property is an array of the environments you want to disable a feature in, e.g. `prod`.
+
+## Using a feature flag
+
+Feature flags can be used by the `*dhFeatureflag` directive or the `DhFeatureFlagsService.`
+
+<mark>Notice: Unknown feature flags not added to the feature flag configuration are treated as enabled.</mark>
```
[the remark this answers is not in the document]
Comment (frontend-specialisten) [reply libs/dh/shared/feature-flags/README.md:31]: @ManBearTM You're absolutely right unless you "cheat" with `as DhFeatureFlags`. This is ofc. not the intended usage but the behavior would be as mentioned in the doc. However, we could skip documenting this, as it's not intended usage what do you think?

[the remark this answers is not in the document]
Comment (frontend-specialisten) [reply libs/dh/shared/feature-flags/README.md:31]: Actually there's also the use-case where no feature flags are added to the configuration. then `DhFeatureFlags` will get the type `never`.

File: libs/dh/shared/feature-flags/README.md [hunk 4]
```diff
@@ -1 +1,59 @@
...
+
+```ts
+  constructor(private featureFlagsService: DhFeatureFlagsService) {}
+```
+
+2. Use the `isEnabled` method on the service:
+
+```ts
+this.featureFlagsService.isEnabled('my-awesome-feature-flag');
+```
```
Comment (ManBearTM) [inline libs/dh/shared/feature-flags/README.md:59]: Perhaps add a short section about how to remove the flags again, something like this:
```md
## Remove a feature flag

When it is time to remove a feature flag, it is recommended to begin by removing the entry from
`dhFeatureFlagsConfig`. The compiler will then error in all the files using that feature flag, making
it easier to identify which files need to be cleaned up.
```

Comment (frontend-specialisten) [reply libs/dh/shared/feature-flags/README.md:59]: Good point!

**Review by ManBearTM — APPROVED** [review]

**Review by dzhavat — APPROVED** [review]

**Pull request merged**